### ISSUE TITLE
How to create a singleton for bramqp?

### ISSUE THREAD

**Issue by jhnferraris** [bakkerthehacker/bramqp#53]
Hello! I'm new to this library and just want to ask if it is possible to reuse the initialize anywhere in the nodejs code?

I understand that when setting up `bramqp` the minimum is this

```
var bramqp = require('bramqp');
var net = require('net');

var socket = net.connect({
  port: 5672
})

bramqp.initialize(socket, 'rabbitmq/full/amqp0-9-1.stripped.extended', function(initError, handle) {
    if (initError) {
        console.log(initError);
    }
});
```

My goal is to reuse the instance of `bramqp` and just use `handle` to allow me execute `handle.basic.consume` anywhere.

Thanks for the help!

**Comment by bakkerthehacker** [comment]
TLDR: just call `initialize` and `openAMQPCommunication` once each and then pass the `handle ` around

Hi!  I'm not sure exactly what your use case is here, so correct me on anything i guess wrong here.  

`bramqp` is the library module.  You could `require('bramqp')` in multiple locations in your code base and call `initialize` and `openAMQPCommunication` multiple times, one for each location.  While this technically works, it is very wasteful.

A better solution would be to call `bramqp.initialize` and `handle.openAMQPCommunication` only once.  After calling these, the handle can be passed around your code and things like `basic.consume` on channel `1` should work properly wherever it is used.  

Something important to keep in mind is that there are combinations of methods, like `basic.produce` followed by `content`, that cannot have any other method sent between them.  Depending on how you have your callbacks setup, it is possibly to incorrectly send some other method between these.

If this is the case, it may be useful to open some more channels throughout your code, other than channel `1`, so that if `basic.produce` followed by `content` happens on channel `1`, other communication can happen on channel `2` without interrupting channel `1`.

Another solution is to only use one channel and to pass the `content` directly to the `basic.produce` method. In this instance, bramqp will ensure that the content comes immediately next.

Using more channels becomes more important if your codebase uses things like `cluster` or `child_process`.  In these situations, the workers would also get a handle like before but each worker would open their own channel to ensure they don't cause issues with any other workers.  It is possible in these situations to not even call `handle.openAMQPCommunication`, but instead have something more complicated instead.  However, the `bramqp.initialize` is still called only once.  

I hope this has helped explain some of the different use cases for bramqp.  I am going to close this issue, but if you still have more questions feel free to reopen it or open a new one.
